src/composables/sdk.ts

```diff
@@ -0,0 +1,66 @@
+import { ref } from 'vue';
+import type {
+  IDefaultComposableOptions,
+  Protocol,
+} from '@/types';
+import { BaseProtocol } from '@/protocols/BaseProtocol';
+import { AeternityProtocol } from '@/protocols/aeternity/AeternityProtocol';
+import { BitcoinProtocol } from '@/protocols/bitcoin/BitcoinProtocol';
+import { DogecoinProtocol } from '@/protocols/dogecoin/DogecoinProtocol';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+  PROTOCOL_DOGECOIN,
+} from '@/popup/utils';
+import { watchUntilTruthy } from '@/utils';
+import { useAccounts } from './accounts';
+
+const isSdkRegistryReady = ref(true);
+const sdkRegistry: Partial<Record<Protocol, BaseProtocol>> = {};
+
+export function useSdk({ store }: IDefaultComposableOptions) {
+  const { activeAccount } = useAccounts({ store });
+
+  function registerSdk(protocol: Protocol, instance: BaseProtocol) {
+    if (sdkRegistry[protocol]) {
+      throw new Error('Protocol instance already exists in the registry');
+    }
+    sdkRegistry[protocol] = instance;
+  }
+
+  function getSdkFromRegistry(protocol: Protocol): BaseProtocol | undefined {
+    return sdkRegistry[protocol];
+  }
+
+  function initSdk(protocol: Protocol): BaseProtocol {
+    switch (protocol) {
+      case PROTOCOL_AETERNITY:
+        return new AeternityProtocol(store);
+      case PROTOCOL_BITCOIN:
+        return new BitcoinProtocol();
+      case PROTOCOL_DOGECOIN:
+        return new DogecoinProtocol();
```

Comment:
Instead of building large switch...case I suggest to first create a constant that would store references to all classes, eg.:
```
const protocolClassReferences: Record<Protocol, BaseProtocol> = {
  aeternity: AeternityProtocol,
  ...
};
```
and then returning it:
```
const protocolClass = protocolClassReferences[protocol];
if (protocolClass) {
  return new protocolClass(store);
} else {
  throw...
}
```
I see that not all protocol classes requires the store argument, but I think the abstract class should force not only the same internals, but also the arguments. Soon we will remove the Vuex so it won't be a problem.